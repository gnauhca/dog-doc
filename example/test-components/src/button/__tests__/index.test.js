import React from 'react';
import { mount, render, shallow } from 'enzyme';
import Button from '..';
import Icon from '../../icon';

describe('Button', () => {

  it('renders correctly', () => {
    const wrapper = render(<Button>button text</Button>);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders theme correctly', () => {
    let theme = 'default';
    let wrapper;
    wrapper = shallow(<Button>button text</Button>);
    expect(wrapper.find('button').hasClass(/type-default/)).toBeTruthy();

    wrapper = shallow(<Button theme={theme}>button text</Button>);
    expect(wrapper.find('button').hasClass(/type-default/)).toBeTruthy();

    theme = 'primary';
    wrapper = shallow(<Button theme={theme}>button text</Button>);
    expect(wrapper.find('button').hasClass(/type-primary/)).toBeTruthy();
  });

  it('renders size correctly', () => {
    let size = 'default';
    let wrapper;
    wrapper = shallow(<Button>button text</Button>);
    expect(wrapper.find('button').hasClass(/size-default/)).toBeTruthy();

    wrapper = shallow(<Button size={size}>button text</Button>);
    expect(wrapper.find('button').hasClass(/size-default/)).toBeTruthy();

    size = 'large';
    wrapper = shallow(<Button size={size}>button text</Button>);
    expect(wrapper.find('button').hasClass(/size-large/)).toBeTruthy();
  });

  it('supports specify icon by string', () => {
    const wrapper = shallow(<Button icon="delete">button text</Button>);
    const wrapperSuffix = shallow(<Button suffixIcon="delete">button text</Button>);
    expect(wrapper).toMatchSnapshot();
    expect(wrapperSuffix).toMatchSnapshot();
  });

  it('supports specify icon by node', () => {
    const wrapper1 = shallow(<Button icon={<Icon type="delete" />}>button text</Button>);
    const wrapper2 = shallow(<Button icon="delete">button text</Button>);

    const wrapperSuffix1 = shallow(<Button suffixIcon={<Icon type="delete" />}>button text</Button>);
    const wrapperSuffix2 = shallow(<Button suffixIcon="delete">button text</Button>);

    expect(wrapper1.html()).toEqual(wrapper2.html());
    expect(wrapperSuffix1.html()).toEqual(wrapperSuffix2.html());
  });

  it('renders loading icon when loading', () => {
    const wrapper = shallow(<Button loading={true} icon={<Icon type="delete" />}>button text</Button>);
    expect(wrapper).toMatchSnapshot();
  });

  it('shouldn\'t responses click when loading or disabled', () => {
    const onClick = jest.fn();
    const wrapper = mount(<Button onClick={onClick}>button text</Button>);

    wrapper.find('button').simulate('click');
    expect(onClick).toHaveBeenCalledTimes(1);
    wrapper.setProps({ disabled: true });

    onClick.mockClear();
    wrapper.find('button').simulate('click');
    expect(onClick).toHaveBeenCalledTimes(0);
  });

  it('support old theme api "type"', () => {
    const wrapper = mount(<div>
      <Button type="primary">主按钮</Button>
      <Button type="default">次按钮</Button>
      <Button type="text">文字按钮</Button>

      <Button type="submit">submit</Button>
    </div>);

    expect(wrapper).toMatchSnapshot();
  });
});
