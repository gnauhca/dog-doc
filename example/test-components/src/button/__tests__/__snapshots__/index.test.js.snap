// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  class="ten-button ten-button--size-default ten-button--type-default"
  type="button"
>
  <span>
    button text
  </span>
</button>
`;

exports[`Button renders loading icon when loading 1`] = `
<button
  className="ten-button ten-button--loading ten-button--size-default ten-button--type-default"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <Icon
    type="loading_gradient"
  />
  <span>
    button text
  </span>
</button>
`;

exports[`Button support old theme api "type" 1`] = `
<div>
  <Button
    autoWidth={false}
    block={false}
    disabled={false}
    loading={false}
    round={false}
    size="default"
    theme="default"
    type="primary"
  >
    <button
      className="ten-button ten-button--size-default ten-button--type-primary"
      disabled={false}
      onClick={[Function]}
    >
      <span>
        主按钮
      </span>
    </button>
  </Button>
  <Button
    autoWidth={false}
    block={false}
    disabled={false}
    loading={false}
    round={false}
    size="default"
    theme="default"
    type="default"
  >
    <button
      className="ten-button ten-button--size-default ten-button--type-default"
      disabled={false}
      onClick={[Function]}
    >
      <span>
        次按钮
      </span>
    </button>
  </Button>
  <Button
    autoWidth={false}
    block={false}
    disabled={false}
    loading={false}
    round={false}
    size="default"
    theme="default"
    type="text"
  >
    <button
      className="ten-button ten-button--size-default ten-button--type-text"
      disabled={false}
      onClick={[Function]}
    >
      <span>
        文字按钮
      </span>
    </button>
  </Button>
  <Button
    autoWidth={false}
    block={false}
    disabled={false}
    loading={false}
    round={false}
    size="default"
    theme="default"
    type="submit"
  >
    <button
      className="ten-button ten-button--size-default ten-button--type-default"
      disabled={false}
      onClick={[Function]}
      type="submit"
    >
      <span>
        submit
      </span>
    </button>
  </Button>
</div>
`;

exports[`Button supports specify icon by string 1`] = `
<button
  className="ten-button ten-button--size-default ten-button--type-default"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <Icon
    type="delete"
  />
  <span>
    button text
  </span>
</button>
`;

exports[`Button supports specify icon by string 2`] = `
<button
  className="ten-button ten-button--size-default ten-button--type-default"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <span>
    button text
  </span>
  <Icon
    type="delete"
  />
</button>
`;
